File: src/main/java/me/narutopig/bors/enchantments/enchantments/ExperienceEnchant.java
package me.narutopig.bors.enchantments.enchantments;

import me.narutopig.bors.CustomEnchants;
import org.bukkit.enchantments.Enchantment;
import org.bukkit.entity.ExperienceOrb;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerExpChangeEvent;
import org.bukkit.event.player.PlayerItemMendEvent;
import org.bukkit.inventory.ItemStack;

import java.util.Map;

public class ExperienceEnchant implements Listener {
    @EventHandler
    public void onMending(PlayerItemMendEvent event) {
        ExperienceOrb experienceOrb = event.getExperienceOrb();
        Player player = event.getPlayer();

        ItemStack hand = player.getInventory().getItemInMainHand();

        int level = hand.getEnchantments().getOrDefault(CustomEnchants.EXPERIENCE, 0);

        player.giveExp(experienceOrb.getExperience() * level);
    }

    @EventHandler
    public void onExpChange(PlayerExpChangeEvent event) {
        int amount = event.getAmount();
        Player player = event.getPlayer();
        ItemStack hand = player.getInventory().getItemInMainHand();
        if (amount > 0) {
            Map<Enchantment, Integer> enchants = hand.getEnchantments();
            player.giveExp(amount * enchants.getOrDefault(CustomEnchants.EXPERIENCE, 0));
        }
    }
}

File: src/main/kotlin/io/github/narutopig/enchantments/ExperienceEnchant.kt
package io.github.narutopig.enchantments

import io.github.narutopig.CustomEnchants
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerExpChangeEvent

class ExperienceEnchant : Listener {
    @EventHandler
    fun onExpChange(event: PlayerExpChangeEvent) {
        val amount = event.amount
        val player = event.player
        val hand = player.inventory.itemInMainHand
        if (amount > 0) {
            val enchants = hand.enchantments
            player.giveExp(amount * enchants.getOrDefault(CustomEnchants.EXPERIENCE, 0))
        }
    }
}